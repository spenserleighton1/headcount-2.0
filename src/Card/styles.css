.card {
  height: 300px;
  width: 250px;
  display: inline-block;
  text-align: center;
  border: 1px solid gray;
  margin: 30px;
}

ul {
  padding: 0;
  list-style-type: none;
}

.select {
  border: 3px solid blue;
}