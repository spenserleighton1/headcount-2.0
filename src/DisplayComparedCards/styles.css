.comparedCards {
  float: right;
  width: 250px;
  border: 2px solid blue;
}