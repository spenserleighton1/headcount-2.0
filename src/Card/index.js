import React from 'react';
import './styles.css'
import PropTypes from 'prop-types';


const Card = (props) => {
  return(
    <div className={`card${props.select ? ' select' : ''}`}
         onClick={()=> { props.findDistrictByClick(props.location) }}>
    <h3>{props.location}</h3>
    <ul>
    {Object.keys(props.stats).map((stat, i) => {
      return <li 
              key={i}
              style={props.stats[stat] >= 0.5 ? {color: 'red'} : {color: 'green'}}>
              {stat}: {props.stats[stat]}
              </li>})}
    </ul>
    </div>
    )
}

export default Card;
