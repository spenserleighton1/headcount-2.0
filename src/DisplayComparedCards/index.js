import React from 'react';
import Card from '../Card';
import PropTypes from 'prop-types';
import './styles.css'

const DisplayComparedCards = (props) => {
  const comparedCards = Object.keys(props.data).map((district, i) => {
    return <Card 
              location={ props.data[district].location }
              stats={ props.data[district].stats }
              key={ i } 
              select={ props.data[district].selected }
              findDistrictByClick={ props.findDistrictByClick }
              compareDistrictAverages={ props.compareDistrictAverages }
          />
  })
   
  const dataToCompare = Object.keys(props.data)


  if(dataToCompare.length === 2) {
    const data = props.compareDistrictAverages(dataToCompare[0], dataToCompare[1])
    return(
        <div>
          { comparedCards }
          <div className='comparedCards'>
            <p> Comparison: </p>
            <p>{ dataToCompare[0] }: { data[dataToCompare[0]] }</p>
            <p>{ dataToCompare[1] }: { data[dataToCompare[1]] }</p>
            <p>{ 'Compared' }: { data['compared'] }</p>
          </div>
        </div>
      )
  }
    
  return(
    <div>
      { comparedCards }
    </div>
    )
}

Card.propTypes = {

}

export default DisplayComparedCards;