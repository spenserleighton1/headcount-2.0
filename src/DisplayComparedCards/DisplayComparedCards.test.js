import React from 'react';
import DisplayComparedCards from './index.js'
import { shallow } from 'enzyme';
import Card from '../Card'

describe('DisplayComparedCards', () => {
  let wrapper;
  let mockData;
  let mockFn;

  beforeEach(() => {
    mockData = { COLORADO: 
         { location: 'COLORADO',
           selected: true,
           stats: 
            { '2004': 0.24,
              '2005': 0.278,
              '2006': 0.337,
              '2007': 0.395,
              '2008': 0.536,
              '2009': 0.598,
              '2010': 0.64,
              '2011': 0.672,
              '2012': 0.695,
              '2013': 0.703,
              '2014': 0.741 } },
        'ACADEMY 20': 
         { location: 'ACADEMY 20',
           selected: true,
           stats: 
            { '2004': 0.302,
              '2005': 0.267,
              '2006': 0.354,
              '2007': 0.392,
              '2008': 0.385,
              '2009': 0.39,
              '2010': 0.436,
              '2011': 0.489,
              '2012': 0.479,
              '2013': 0.488,
              '2014': 0.49 } } }


  mockFn = jest.fn()
  wrapper = shallow(<DisplayComparedCards data={ mockData }
                                          compareDistrictAverages={ mockFn }
                                          findDistrictByClick={ mockFn } 
                                          />)

  })
  it('matches the Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // it('should render the correct amount of cards', () => {
  //   expect(wrapper.find(Card).length).toEqual(2)
  // })
})