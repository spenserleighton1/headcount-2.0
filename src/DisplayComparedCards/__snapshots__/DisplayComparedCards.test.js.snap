// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayComparedCards matches the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayComparedCards
    data={
        Object {
            "ACADEMY 20": Object {
              "location": "ACADEMY 20",
              "stats": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
            },
            "COLORADO": Object {
              "location": "COLORADO",
              "stats": Object {
                "2004": 0.24,
                "2005": 0.278,
                "2006": 0.337,
                "2007": 0.395,
                "2008": 0.536,
                "2009": 0.598,
                "2010": 0.64,
                "2011": 0.672,
                "2012": 0.695,
                "2013": 0.703,
                "2014": 0.741,
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          findDistrictByClick={undefined}
          location="COLORADO"
          select={undefined}
          stats={
                    Object {
                              "2004": 0.24,
                              "2005": 0.278,
                              "2006": 0.337,
                              "2007": 0.395,
                              "2008": 0.536,
                              "2009": 0.598,
                              "2010": 0.64,
                              "2011": 0.672,
                              "2012": 0.695,
                              "2013": 0.703,
                              "2014": 0.741,
                            }
          }
/>,
        <Card
          findDistrictByClick={undefined}
          location="ACADEMY 20"
          select={undefined}
          stats={
                    Object {
                              "2004": 0.302,
                              "2005": 0.267,
                              "2006": 0.354,
                              "2007": 0.392,
                              "2008": 0.385,
                              "2009": 0.39,
                              "2010": 0.436,
                              "2011": 0.489,
                              "2012": 0.479,
                              "2013": 0.488,
                              "2014": 0.49,
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "findDistrictByClick": undefined,
          "location": "COLORADO",
          "select": undefined,
          "stats": Object {
            "2004": 0.24,
            "2005": 0.278,
            "2006": 0.337,
            "2007": 0.395,
            "2008": 0.536,
            "2009": 0.598,
            "2010": 0.64,
            "2011": 0.672,
            "2012": 0.695,
            "2013": 0.703,
            "2014": 0.741,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "findDistrictByClick": undefined,
          "location": "ACADEMY 20",
          "select": undefined,
          "stats": Object {
            "2004": 0.302,
            "2005": 0.267,
            "2006": 0.354,
            "2007": 0.392,
            "2008": 0.385,
            "2009": 0.39,
            "2010": 0.436,
            "2011": 0.489,
            "2012": 0.479,
            "2013": 0.488,
            "2014": 0.49,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            findDistrictByClick={undefined}
            location="COLORADO"
            select={undefined}
            stats={
                        Object {
                                    "2004": 0.24,
                                    "2005": 0.278,
                                    "2006": 0.337,
                                    "2007": 0.395,
                                    "2008": 0.536,
                                    "2009": 0.598,
                                    "2010": 0.64,
                                    "2011": 0.672,
                                    "2012": 0.695,
                                    "2013": 0.703,
                                    "2014": 0.741,
                                  }
            }
/>,
          <Card
            findDistrictByClick={undefined}
            location="ACADEMY 20"
            select={undefined}
            stats={
                        Object {
                                    "2004": 0.302,
                                    "2005": 0.267,
                                    "2006": 0.354,
                                    "2007": 0.392,
                                    "2008": 0.385,
                                    "2009": 0.39,
                                    "2010": 0.436,
                                    "2011": 0.489,
                                    "2012": 0.479,
                                    "2013": 0.488,
                                    "2014": 0.49,
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "findDistrictByClick": undefined,
            "location": "COLORADO",
            "select": undefined,
            "stats": Object {
              "2004": 0.24,
              "2005": 0.278,
              "2006": 0.337,
              "2007": 0.395,
              "2008": 0.536,
              "2009": 0.598,
              "2010": 0.64,
              "2011": 0.672,
              "2012": 0.695,
              "2013": 0.703,
              "2014": 0.741,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "findDistrictByClick": undefined,
            "location": "ACADEMY 20",
            "select": undefined,
            "stats": Object {
              "2004": 0.302,
              "2005": 0.267,
              "2006": 0.354,
              "2007": 0.392,
              "2008": 0.385,
              "2009": 0.39,
              "2010": 0.436,
              "2011": 0.489,
              "2012": 0.479,
              "2013": 0.488,
              "2014": 0.49,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
